test/extended/util/proxy.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	"context"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HasProxy returns whether or not the currently installed cluster has a proxy configured
+func HasProxy(oc *CLI) (bool, error) {
+	proxy, err := oc.AdminConfigClient().ConfigV1().Proxies().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return false, err
+	} else if kerrors.IsNotFound(err) {
+		return false, nil
+	}
```

Comment:
Looks like `kerrors.IsNotFound` returns `false` if the input is nil, so we can simplify a bit:

```suggestion
	if kerrors.IsNotFound(err) {
		return false, nil
	}
	if err != nil {
		return false, err
	}
```

Reply:
updated in next push